includes/class-sensei-analysis.php

```diff
@@ -45,9 +45,77 @@ public function __construct( $file ) {
 			add_action( 'admin_init', array( $this, 'report_download_page' ) );
 
 			add_filter( 'user_search_columns', array( $this, 'user_search_columns_filter' ), 10, 3 );
+
+			// Add custom navigation.
+			add_action( 'in_admin_header', [ $this, 'add_custom_navigation' ] );
+		}
+	}
+
+	/**
+	 * Add custom navigation to the admin pages.
+	 *
+	 * @since 4.2.0
+	 * @access private
+	 */
+	public function add_custom_navigation() {
+		// phpcs:ignore WordPress.Security.NonceVerification -- No action, nonce is not required.
+		$is_reports_page = isset( $_GET['page'] ) && ( $_GET['page'] === $this->page_slug );
+
+		if ( ! $is_reports_page ) {
+			return;
+		}
+
+		// phpcs:ignore WordPress.Security.NonceVerification -- No action, nonce is not required.
+		if ( isset( $_GET['course_id'] ) || isset( $_GET['lesson_id'] ) || isset( $_GET['user_id'] ) ) {
+			return;
```

Comment:
Just wondering - how the navigation will be handled on subscreens? Is there going to be an update in a separate PR or we are just leaving the current nav?

Reply:
I've found the answer in a previous comment. 🙂